src/App.tsx

```diff
@@ -1,26 +1,393 @@
+  if (completedStatus === null) {
+    return todos;
+  }
+
+  return todos.filter(todo => todo.completed === completedStatus);
+}
+
+function removeTodoById(todos: Todo[], id: number) {
+  return todos.filter(todo => todo.id !== id);
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Error | null>(null);
+  const [activeFilter, setActiveFilter] = useState<Filter>(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<number[] | null>(null);
+
+  const [inputText, setInputText] = useState<string>('');
+  const [isDeletingTodo, setIsDeletingTodo] = useState<boolean>(false);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const addTodoField = useRef<HTMLInputElement>(null);
+  const errorTimeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const isAllCompleted = !todos.some(todo => todo.completed === false);
+  const hasCompleted = todos.some(todo => todo.completed === true);
+
+  const activeCount: number = todos.reduce((acc, todo) => {
+    if (todo.completed === false) {
+      return acc + 1;
+    }
+
+    return acc;
+  }, 0);
+  const filterValues = Object.values(Filter);
+
+  function ShowError(message: Error) {
+    if (errorTimeoutRef.current) {
+      clearTimeout(errorTimeoutRef.current);
+    }
+
+    setErrorMessage(message);
+    errorTimeoutRef.current = setTimeout(() => setErrorMessage(null), 3000);
+  }
+
+  function changeStatus(
+    id: number,
+    todosState: Dispatch<SetStateAction<Todo[]>>,
+  ) {
+    todosState(prev => {
+      return prev.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      );
+    });
+  }
+
+  function handleTodoStatusChange(id: number) {
+    changeStatus(id, setFilteredTodos);
+    changeStatus(id, setTodos);
+  }
+
+  useEffect(() => {
+    client
+      .get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+        setFilteredTodos(fetchedTodos);
+        if (addTodoField.current !== null) {
+          addTodoField.current.focus();
+        }
+      })
+      .catch(() => ShowError(Error.LoadError));
+  }, []);
+
+  useEffect(() => {
+    if (addTodoField.current !== null && tempTodo === null) {
+      /*tempTodo === null для того, не виконувати це два рази (бо стейт tempTodo спочатку змінюється на об'єкт а потім змінюється на null)*/
+      addTodoField.current.focus();
+    }
+  }, [tempTodo]);
+
+  function handleFilter(filterParam: Filter) {
+    let filter;
+
+    if (filterParam === Filter.All) {
+      filter = null;
+    } else if (filterParam === Filter.Active) {
+      filter = false;
+    } else {
+      filter = true;
+    }
+
+    setFilteredTodos(filterTodosByStatus(todos, filter));
+    setActiveFilter(filterParam);
+  }
```

Comment:
u can use ur enum in this case

```suggestion
  function handleFilter(filterParam: Filter) {
    setFilteredTodos(filterTodosByStatus(todos, filterParam));
    setActiveFilter(filterParam);
  }
```